### PR TITLE
Update app badge when we get badged +plus random fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/chat/inbox.js [hunk 1]
```diff
@@ -255,12 +255,12 @@ function * unboxConversations (conversationIDKeys: Array<Constants.ConversationI
 
       switch (error.typ) {
         case ChatTypes.LocalConversationErrorType.selfrekeyneeded: {
-          yield call(Creators.updateInbox(conversation))
+          yield put(Creators.updateInbox(conversation))
```
Comment (chrisnojima) [inline shared/actions/chat/inbox.js]: these are puts, not calls

File: shared/actions/chat/index.js [hunk 2]
```diff
@@ -805,7 +805,9 @@ function * _badgeAppForChat (action: Constants.BadgeAppForChat): SagaGenerator<a
     const unread = conv.get('UnreadMessages') > 0
     const selected = (Constants.conversationIDToKey(conv.get('convID')) === selectedConversationIDKey)
     const addThisConv = (unread && (!selected || !windowFocused))
-    return addThisConv ? acc + 1 : acc
+    // Desktop shows number of thread unread. Mobile shows number of messages unread
```
Comment (chrisnojima) [inline shared/actions/chat/index.js]: badge the actual count, not per thread

File: shared/chat/inbox/index.native.js [hunk 3]
```diff
@@ -100,7 +100,7 @@ const BottomLine = ({participantNeedToRekey, youNeedToRekey, isMuted, showBold,
 
   return (
     <Box style={{...globalStyles.flexBoxRow, backgroundColor, flexGrow: 1, maxHeight: 16, minHeight: 16, position: 'relative'}}>
-      <Box style={{...globalStyles.flexBoxRow, aligntItems: 'flex-start', bottom: 0, justifyContent: 'flex-start', left: 0, position: 'absolute', right: 0, top: 0}}>
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'flex-start', bottom: 0, justifyContent: 'flex-start', left: 0, position: 'absolute', right: 0, top: 0}}>
```
Comment (chrisnojima) [inline shared/chat/inbox/index.native.js]: typo

File: shared/main-shared.native.js [hunk 4]
```diff
@@ -12,10 +13,24 @@ import {getUserImageMap, loadUserImageMap} from './util/pictures'
 import {initAvatarLookup, initAvatarLoad} from './common-adapters'
 import {listenForNotifications} from './actions/notifications'
 import {persistRouteState, loadRouteState} from './actions/platform-specific.native'
-import {setRouteState, navigateUp} from './actions/route-tree'
+import {setRouteState} from './actions/route-tree'
 
 // TODO type this later
-type Props = any
+type Props = {
```
Comment (chrisnojima) [inline shared/main-shared.native.js]: more typing

File: shared/main-shared.native.js [hunk 5]
```diff
@@ -40,6 +55,10 @@ class Main extends Component<void, any, void> {
     if (this.props.routeState !== nextProps.routeState) {
       this._persistRoute()
     }
+
+    if (this.props.menuBadgeCount !== nextProps.menuBadgeCount) {
```
Comment (chrisnojima) [inline shared/main-shared.native.js]: set the badge

File: shared/main-shared.native.js [hunk 6]
```diff
@@ -67,29 +86,28 @@ class Main extends Component<void, any, void> {
...
     favorite: {privateBadge, publicBadge},
     push: {permissionsPrompt},
     routeTree: {routeDef, routeState},
+    notifications: {menuBadgeCount},
   }) => ({
     dumbFullscreen,
     folderBadge: privateBadge + publicBadge,
+    menuBadgeCount,
     mountPush: extendedConfig && !!extendedConfig.defaultDeviceID && loggedIn && bootStatus === 'bootStatusBootstrapped',
-    permissionsPrompt,
```
Comment (chrisnojima) [inline shared/main-shared.native.js]: not used

File: shared/nav.native.js [hunk 7]
```diff
@@ -51,7 +51,7 @@ class CardStackShim extends Component {
         routes: stack.map(route => {
           const routeName = route.path.join('/')
           return {key: routeName, routeName, params: route}
-        }),
+        }).toArray(),
```
Comment (chrisnojima) [inline shared/nav.native.js]: this needs to be not immutable

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 8]
```diff
@@ -49,7 +49,7 @@ - (void) setupGo
 #endif
 
   BOOL securityAccessGroupOverride = isSimulator;
-  BOOL skipLogFile = isDebug;
+  BOOL skipLogFile = false;
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: this hoses xcode otherwise

File: shared/settings/nav/index.native.js [hunk 9]
```diff
@@ -71,7 +71,6 @@ const itemStyle = {
   paddingRight: globalMargins.small,
   alignItems: 'center',
   position: 'relative',
-  textTransform: 'uppercase',
```
Comment (chrisnojima) [inline shared/settings/nav/index.native.js]: not supported on mobile

File: shared/main-shared.native.js [hunk 10]
```diff
@@ -12,10 +13,24 @@ import {getUserImageMap, loadUserImageMap} from './util/pictures'
 import {initAvatarLookup, initAvatarLoad} from './common-adapters'
 import {listenForNotifications} from './actions/notifications'
 import {persistRouteState, loadRouteState} from './actions/platform-specific.native'
-import {setRouteState, navigateUp} from './actions/route-tree'
+import {setRouteState} from './actions/route-tree'
 
 // TODO type this later
```
Comment (MarcoPolo) [inline shared/main-shared.native.js]: ✂️

File: shared/reducers/notifications.js [hunk 11]
```diff
@@ -56,7 +57,10 @@ export default function (state: State = initialState, action: NotificationAction
         menuNotifications.chatBadge = badgeAction.payload.count || 0
       }
 
-      const menuBadgeCount = Object.keys(menuNotifications).reduce((total, n) => total + menuNotifications[n], 0)
+      // Menu badge is chat only currently
+      const menuBadgeCount = isMobile
```
Comment (MarcoPolo) [inline shared/reducers/notifications.js]: Was this really needed? did we get notifications for non chat stuff?

Comment (chrisnojima) [inline shared/reducers/notifications.js]: we do, didn't want to touch too much of this code

File: shared/main-shared.native.js [hunk 12]
```diff
@@ -12,10 +13,23 @@ import {getUserImageMap, loadUserImageMap} from './util/pictures'
...
+import {setRouteState} from './actions/route-tree'
 
-// TODO type this later
-type Props = any
+type Props = {
+  dumbFullscreen: boolean,
+  folderBadge: number,
+  menuBadgeCount: number,
+  mountPush: boolean,
+  routeDef: any,
```
Comment (chromakode) [inline shared/main-shared.native.js]: These have types in https://github.com/keybase/client/blob/bf110bbc1503c6514227fb5b860436dbe95f5dde/shared/route-tree/render-route.js#L33

File: shared/main-shared.native.js [hunk 13]
```diff
@@ -12,10 +13,23 @@ import {getUserImageMap, loadUserImageMap} from './util/pictures'
...
+  routeState: any,
+  showPushPrompt: any,
+  bootstrap: () => void,
+  hello: () => void,
+  listenForNotifications: () => void,
+  loadRouteState: () => void,
+  persistRouteState: () => void,
+  setRouteState: (path: any, partialState: any) => void,
```
Comment (chromakode) [inline shared/main-shared.native.js]: `path` should be `I.List<string>`

File: shared/main-shared.native.js [hunk 14]
```diff
@@ -12,10 +13,23 @@ import {getUserImageMap, loadUserImageMap} from './util/pictures'
...
-// TODO type this later
-type Props = any
+type Props = {
+  dumbFullscreen: boolean,
+  folderBadge: number,
+  menuBadgeCount: number,
+  mountPush: boolean,
+  routeDef: any,
+  routeState: any,
+  showPushPrompt: any,
```
Comment (chromakode) [inline shared/main-shared.native.js]: `boolean`?

File: shared/main-shared.native.js (hunk 13 above)
Comment (chrisnojima) [inline shared/main-shared.native.js]: I just partially typed this. before it had nothing

**Pull request merged**